### PR TITLE
Use ProxyReader contract

### PR DESCRIPTION
**Pull request by hivtushok**
In this PR:
- Replacement of [Registry](https://etherscan.io/address/0xD1E5b0FF1287aA9f9A268759062E4Ab08b9Dacbe#code) & [Resolver](https://etherscan.io/address/0xb66DcE2DA6afAAa98F2013446dBCB0f4B0ab2842#code) contracts with the usage of [ProxyReader](https://etherscan.io/address/0x7ea9Ee21077F84339eDa9C80048ec6db678642B1#code) instead;
- Code cleanups;
- Bump to v1.2.0.

### CODE REVIEW

**Comment by hivtushok** [comment]
[WIP]: pending offline discussion with @bogdan regarding the ABIs.

File: src/test/java/com/unstoppabledomains/ProxyReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ProxyReaderTest {
+
+    private static final String URL = "https://mainnet.infura.io/v3/781c1e5cae32417b93eac26042950d25";
```
Comment (merenkoff) [inline src/test/java/com/unstoppabledomains/ProxyReaderTest.java:13]: seems like we commit a token of `mainnet.infura.io` Is this OK for open source solution?
may be better to use link similar as we use in resolution swift

`https://main-rpc.linkpool.io`

File: build.gradle [hunk 2]
```diff
@@ -18,7 +18,7 @@ targetCompatibility = 1.8
 
 group = "com.unstoppabledomains.resolution"
 archivesBaseName = "resolution"
-version = "1.0.9"
+version = "1.1.0"
```
Comment (JohnnyJumper) [inline build.gradle:21]: I just double-checked v1.1.0 is already published. Let's update it to 1.2.0

File: .gitignore [hunk 3]
```diff
@@ -1,33 +1,68 @@
...
+dist/
+nbdist/
+nbactions.xml
+nb-configuration.xml
+
+# Visual Studio Code
+.vscode/
+.code-workspace
 
-gradle.properties
```
Comment (JohnnyJumper) [inline .gitignore:33]: Let's leave gradle.properties in the gitignore. It does expose our ossrh account and signing password as well

**Review by JohnnyJumper — CHANGES_REQUESTED** [review]
When I ran tests it fails to build for me with this error
Screen Shot 2020-10-05 at 11 44 25 AM

Any ideas why this can happen?

**Comment by JohnnyJumper** [comment]
> When I ran tests it fails to build for me with this error
> Screen Shot 2020-10-05 at 11 44 25 AM
> 
> Any ideas why this can happen?

NVM, doing gradle build fixed it. Looks like when you ran test via VS code it doesn't automatically rebuild the project for you

File: .gitignore (hunk 3 above)
Comment (hivtushok) [reply .gitignore:33]: We should move these credentials into 'repo secrets' when we set up CI/CD. This file usually contains properties for builds, so I think we need it. Let's just avoid committing credentials here for now. Do you agree?

File: build.gradle (hunk 2 above)
Comment (hivtushok) [reply build.gradle]: Ok, I see now, we have two different artifacts at this point: https://search.maven.org/search?q=com.unstoppabledomains
Bumped to v1.2.0 now, thanks!

File: src/test/java/com/unstoppabledomains/ProxyReaderTest.java (hunk 1 above)
Comment (hivtushok) [reply src/test/java/com/unstoppabledomains/ProxyReaderTest.java:13]: Good point, thanks. This will be addressed in the scope of another story.

**Review by JohnnyJumper — APPROVED** [review]

**Review by merenkoff — APPROVED** [review]

**Pull request merged**